src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -673,13 +672,59 @@ void CodeGen::genSSEIntrinsic(GenTreeHWIntrinsic* node)
 void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
 {
     NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
+    GenTree*       op1         = node->gtGetOp1();
+    GenTree*       op2         = node->gtGetOp2();
+    regNumber      targetReg   = node->gtRegNum;
+    var_types      targetType  = node->TypeGet();
     var_types      baseType    = node->gtSIMDBaseType;
-    instruction    ins         = INS_invalid;
+    instruction    ins         = Compiler::insOfHWIntrinsic(intrinsicID, baseType);
+    regNumber      op1Reg      = REG_NA;
+    regNumber      op2Reg      = REG_NA;
+    emitter*       emit        = getEmitter();
+    int            ival        = -1;
 
-    genConsumeOperands(node);
+    if ((op1 != nullptr) && !op1->OperIsList())
+    {
+        op1Reg = op1->gtRegNum;
+        genConsumeOperands(node);
+    }
 
     switch (intrinsicID)
     {
+        // All integer overloads are handled by table codegen
+        case NI_SSE2_CompareLessThan:
```

Comment:
It would be good to combine this with the `Sse` logic, since they should be the same outside of instruction.

Reply:
The same commonalities will be even more visible for scalar operations. IMO this can be done with larger refactoring which would combine all common patterns of SSE, SSE2, SSE3, SSE4.1 intrinsics.

I have created issue to track it #16014

Reply:
There will be even more cases where code could be combined/abstracted for all SSE, SSE2, SSE3, SSE4.1, SSE4.2 hardware intrinsics. From what I have worked on it will be the case for all `Scalar` operations on `double` and later on all integer operations. 

I have created issue to track this #16014 and plan to work on refactoring code in next PRs.

Reply:
Seems I have double comment :)